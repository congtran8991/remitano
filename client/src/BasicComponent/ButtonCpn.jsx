import React from "react";

const ButtonCpn = (props) => {
  const { title = "", onClick = () => {}, style = {}, dataTestId = "defaultTestId" } = props;
  return (
    <button
      data-testid={dataTestId}
      style={style}
      onClick={onClick}
      className="btn"
    >
      {title}
    </button>
  );
};

export default ButtonCpn;
