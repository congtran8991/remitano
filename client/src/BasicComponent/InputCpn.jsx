import React from "react";

const InputCpn = (props) => {
  const {
    value = "",
    placeholder = "",
    onChange = () => {},
    name = "",
    type = "text",
    style = {},
    dataTestId = "defaultTestInputId",
  } = props;
  return (
    <input
      data-testid={dataTestId}
      type={type}
      name={name}
      className="input-cpn"
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      style={style}
    />
  );
};

export default InputCpn;
