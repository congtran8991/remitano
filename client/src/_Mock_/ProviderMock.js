import React from "react";
import { Provider } from "react-redux";
import { store } from "../Redux/store";
import { BrowserRouter } from "react-router-dom";

const providerMock = (props) => {
  return (
    <Provider store={store}>
      <BrowserRouter>{props.children}</BrowserRouter>
    </Provider>
  );
};

export default providerMock;
