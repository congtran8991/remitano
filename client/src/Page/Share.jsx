import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import InputCpn from "../BasicComponent/InputCpn";
import ButtonCpn from "../BasicComponent/ButtonCpn";
import { useNavigate } from "react-router-dom";
import { userLoginSlice } from "../Redux/reducer/userLoginSlice";
import { AXIOS } from "../Utils/axios";
import { getCookie } from "../Utils/client";
import { validateYouTubeUrl } from "../Utils/client";
import { toast } from "react-toastify";
const SharePage = () => {
  const dataUserLogin = useSelector(userLoginSlice);
  const navigate = useNavigate();
  const [urlVideo, setUrlVideo] = useState("");
  useEffect(() => {
    if (!dataUserLogin.isLogin) {
      return navigate("/");
    }
  }, [dataUserLogin.isLogin, navigate]);
  const handleChangeUrl = (e) => {
    setUrlVideo(e.target.value);
  };

  const handleShareUrl = async () => {
    const isValidateUrl = validateYouTubeUrl(urlVideo);
    if (!isValidateUrl)
      return toast.warn("Url không hợp lệ", {
        position: "top-right",
        autoClose: 3000,
        theme: "light",
      });
    try {
      const refreshToken = getCookie("refresh_token");
      const param = {
        path: "/movie/addListMovie",
        token: refreshToken,
        data: {
          url: urlVideo,
          user: {
            userId: dataUserLogin.data.data.id,
            email: dataUserLogin.data.data.email,
          },
        },
        method: "POST",
      };
      await AXIOS(param);
      toast.success("Thêm Url thành công", {
        position: "top-right",
        autoClose: 3000,
        theme: "light",
      });
      navigate("/");
    } catch (error) {
      toast.error(error.message, {
        position: "top-right",
        autoClose: 3000,
        theme: "light",
      });
    }
  };
  return (
    <div className="wrapper-share">
      <div>Youtube URL: &ensp;</div>
      <div>
        <div>
          <InputCpn
            dataTestId="input-url"
            value={urlVideo}
            onChange={handleChangeUrl}
            style={{ height: "30px", width: "300px" }}
          />
        </div>
        <div>
          <ButtonCpn
            dataTestId="btn-shareUrl"
            style={{ height: "30px", width: "305px", marginTop: "5px" }}
            title="Share"
            onClick={handleShareUrl}
          />
        </div>
      </div>
    </div>
  );
};

export default SharePage;
