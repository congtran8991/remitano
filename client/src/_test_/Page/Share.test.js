import { render, fireEvent, screen } from "@testing-library/react";
import Share from "../../Page/Share";
import InputCpn from "../../BasicComponent/InputCpn";
import ButtonCpn from "../../BasicComponent/ButtonCpn";
import ProviderMock from "../../_Mock_/ProviderMock";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("../../Utils/axios", () => ({
  AXIOS: jest.fn().mockResolvedValueOnce([]),
}));

jest.mock("../../Utils/axios", () => ({
  AXIOS: jest.fn().mockRejectedValueOnce({}),
}));

describe("SharePage", () => {
  test("Should handle share video on success", () => {
    render(
      <ProviderMock>
        <Share />
      </ProviderMock>
    );
    const url = "https://www.youtube.com/watch?v=gViRsnlgs6E";
    const inputEmail = screen.getByTestId("input-url");
    fireEvent.change(inputEmail, { target: { value: url } });
    const btnLogin = screen.getByTestId("btn-shareUrl");
    fireEvent.click(btnLogin);
  });

  test("Should handle share video invalid", () => {
    render(
      <ProviderMock>
        <Share />
      </ProviderMock>
    );
    const url = "https:?v=gViRsnlgs6E";
    const inputEmail = screen.getByTestId("input-url");
    fireEvent.change(inputEmail, { target: { value: url } });
    const btnLogin = screen.getByTestId("btn-shareUrl");
    fireEvent.click(btnLogin);
  });
  test("Should update URL video", () => {
    const handleChangeUrl = jest.fn();
    const wrapper = shallow(<InputCpn value="" onChange={handleChangeUrl} />);

    const input = wrapper.find("input");
    const updatedUrl = "https://www.example.com/updated";

    input.simulate("change", { target: { value: updatedUrl } });
    expect(handleChangeUrl).toHaveBeenCalled();
    expect(handleChangeUrl).toHaveBeenCalledWith({
      target: { value: updatedUrl },
    });
  });
});
