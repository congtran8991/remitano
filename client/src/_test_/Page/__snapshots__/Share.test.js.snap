// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Render HomePage 1`] = `undefined`;

exports[`SharePage Render SharePage 1`] = `undefined`;

exports[`SharePage Should update URL video 1`] = `undefined`;
